### ISSUE TITLE
Type mismatch converting json to csv 

### ISSUE THREAD

**Issue by BobbyDimmette** [python-bonobo/bonobo#225]
Running on Windows 7 64-bit with Python 3.6....

>bonobo convert --reader json out.json --writer csv out.csv
TypeError (in JsonReader)
 | Type object argument after ** must be a mapping, not str
 |
 | Traceback (most recent call last):
 |   File "c:\program files\python36\lib\site-packages\bonobo\execution\node.py", line 90, in loop
 |     self.step()
 |   File "c:\program files\python36\lib\site-packages\bonobo\execution\node.py", line 113, in step
 |     self.handle_results(input_bag, input_bag.apply(self._stack))
 |   File "c:\program files\python36\lib\site-packages\bonobo\execution\node.py", line 130, in handle_results
 |     result = next(results)
 |   File "c:\program files\python36\lib\site-packages\bonobo\nodes\io\base.py", line 69, in __call__
 |     yield from self.read(*args, **kwargs)
 |   File "c:\program files\python36\lib\site-packages\bonobo\nodes\io\json.py", line 20, in read
 |     yield self.get_output(line)
 |   File "c:\program files\python36\lib\site-packages\bonobo\nodes\io\base.py", line 34, in get_output
 |     return Bag(**row)
 | TypeError: type object argument after ** must be a mapping, not str

 - JsonReader in=1
 - CsvWriter

>bonobo version
bonobo v.0.5.2 (in c:\program files\python36\lib\site-packages)

**Comment by hartym** [comment]
This does indeed look like a bug, will have a look.

**Comment by hartym** [comment]
@BobbyDimmette Trying to reproduce but "worksforme" (although for now I try on unix, but your error does not sound like windows specific). So 2 things:
- I'm installing a windows VM right now to test this, will take some time though.
- Can you either post your source file or send it privately to me (romain at bonobo dash project dot org) so I can try with your exact source file?
Thanks.

**Comment by hartym** [comment]
@BobbyDimmette Update, I did install a Win10 VM and test a basic CSV to JSON conversion within the machine, using latest stable (0.5.2), and it's working as expected. Please provide your source file so I can dive into what brought this error.

```
PS C:\Users\IEUser\Downloads> bonobo.exe convert Demographic_Statistics_By_Zip_Code.csv Demographic.json
 - CsvReader in=1 out=236
 - JsonWriter in=236 out=236
```

**Comment by BobbyDimmette** [comment]
Thanks.  I sent my JSON file to your private address.

**Comment by hartym** [comment]
Ok.

The good news is that I reproduce your problem using 0.5 on OSX, and also that 0.6 does not have it.

The bad news is that looking at the format of your file (a top-level dictionary), I'm not really sure on how bonobo *should* behave while reading this. Probably it can use .items() somehow on the dict to yield everything, but how would that become a csv? Not really sure. Would be awesome if you could tell me what kind of output format you're looking after, because there is not one obvious way to do the thing, here.

I'm afraid to say that this wont be fixed in 0.5, and that 0.6 (first alpha release today) has it fixed (as in, no exception anymore) but does something by default which is probably not what you want.

I leave this ticket open so we can think a bit more about dict-based json files after release.

About the mimetype detection, I'm not "shocked" about the behaviour, although we have to think about reproducibility of commands. According to python's mimetypes package documentation, used to do this, "on Windows, the current registry settings are loaded" (see https://docs.python.org/3/library/mimetypes.html#mimetypes.init), so that behaviour make sense.

**Comment by hartym** [comment]
Works on 0.6
